data/value_mapping.go

```diff
@@ -0,0 +1,143 @@
+package data
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	jsoniter "github.com/json-iterator/go"
+)
+
+// MappingType see https://github.com/grafana/grafana/blob/main/packages/grafana-data/src/types/valueMapping.ts
+type MappingType string
+
+const (
+	ValueToText  MappingType = "value"
+	RangeToText  MappingType = "range"
+	SpecialValue MappingType = "special"
+)
+
+type SpecialValueMatch string
+
+const (
+	SpecialValueTrue       SpecialValueMatch = "true"
+	SpecialValueFalse      SpecialValueMatch = "false"
+	SpecialValueNull       SpecialValueMatch = "null"
+	SpecialValueNaN        SpecialValueMatch = "nan"
+	SpecialValueNullAndNaN SpecialValueMatch = "null+nan"
+	SpecialValueEmpty      SpecialValueMatch = "empty"
+)
+
+// ValueMappingResult is the results from mapping a value
+type ValueMappingResult struct {
+	Text  string `json:"text,omitempty"`
+	Color string `json:"color,omitempty"`
+	Index int    `json:"index,omitempty"` // just used ofr ui ordering
+}
+
+// ValueMapping allows mapping input values to text and color
+type ValueMapping interface {
+	GetType() MappingType
+}
```

Comment:
If there's truly no case in which any other than the three known types are permitted to fulfill this interface, it's preferable to add an unexported method to the interface. That "seals" the interface, guaranteeing that only the types within the package can ever possibly implement it.

Reply:
nice